src/eth1spec/spec.py

```diff
@@ -333,6 +348,143 @@ def signing_hash(tx: Transaction) -> Hash32:
+    """
+    Computes the hash of a block Header.
+
+    Parameters
+    ----------
+    header : `eth1spec.eth_types.Header`
+        Header of interest.
+
+    Returns
+    -------
+    hash : `eth1spec.eth_types.Hash32`
+        Hash of the Header.
+    """
+    return crypto.keccak256(
+        rlp.encode(
+            (
+                header.parent_hash,
+                header.ommers_hash,
+                header.coinbase,
+                header.state_root,
+                header.transactions_root,
+                header.receipt_root,
+                header.bloom,
+                header.difficulty,
+                header.number,
+                header.gas_limit,
+                header.gas_used,
+                header.timestamp,
+                header.extra_data,
+                header.mix_digest,
+                header.nonce,
+            )
+        )
+    )
+
+
+def get_block_header_by_hash(hash: Hash32, chain: BlockChain) -> Header:
+    """
+    Fetches the block header by searching its hash.
+
+    Parameters
+    ----------
+    hash : `eth1spec.eth_types.Hash32`
+        hash of the header of interest.
+
+    chain : `eth1spec.eth_types.BlockChain`
+        History and current state.
+
+    Returns
+    -------
+    Header : `eth1spec.eth_types.Header`
+        Block header found by its hash.
+    """
+    for block in chain.blocks:
+        if header_hash(block.header) == hash:
+            return block.header
+    else:
+        raise ValueError(f"Could not find Header with hash={hash.hex()}")
+
+
+def validate_proof_of_work(header: Header) -> bool:
+    """
+    Validates the Proof of Work constraints
+
+    Parameters
+    ----------
+    header : `eth1spec.eth_types.Uint`
+        header of interest
+
+    Returns
+    -------
+    verified : `bool`
+        True if Proof of Work constraints are satisfied, False otherwise
+    """
+    # TODO: Implement this method once proof of work
+    #  algorithm is implemented
+    #  https://github.com/quilt/eth1.0-specs/issues/43
+    raise NotImplementedError
+
+
+def validate_gas_limit(gas_limit: Uint, parent_gas_limit: Uint) -> bool:
+    """
+    Validates the gas limit for a block
+
+    Parameters
+    ----------
+    gas_limit : `eth1spec.eth_types.Uint`
+        gas limit of the header of interest
+
+    parent_gas_limit : `eth1spec.eth_types.Uint`
+        gas limit of the porent header of interest
+
+    Returns
+    -------
+    verified : `bool`
+        True if header gas limit constraints are satisfied, False otherwise
+    """
+    max_adjustment_delta = parent_gas_limit // GAS_LIMIT_ADJUSTMENT_FACTOR
+    if gas_limit >= parent_gas_limit + max_adjustment_delta:
+        return False
+    if gas_limit <= parent_gas_limit - max_adjustment_delta:
+        return False
+    if gas_limit < GAS_LIMIT_MINIMUM:
+        return False
+
+    return True
+
+
+def calculate_block_difficulty(header: Header, parent_header: Header) -> Uint:
```

Comment:
Seems like you only need the parent's number, timestamp, difficult, and the current timestamp -- is there a reason to not just accept those parameters instead both headers?

Reply:
I think this is probably a good pattern (passing in explicit values as opposed to the full header object).  This avoids complex future typing problems where you end up trying to retain this function signature and ending up with complex types for your `Header` definition.  Keeps you from "pretending" that some of these validation functions actually have common signatures across different forks.